### ISSUE TITLE
SNAP format support？

### ISSUE THREAD

**Issue by rogerogerogeroger** [vincentbernat/lldpd#340]
Hello, Vincent
Is it possible to configure the lldp protocol to transmit in the SNAP format?

**Comment by vincentbernat** [comment]
No, there is currently nothing for that. Are you using a non-Ethernet network?

**Comment by rogerogerogeroger** [comment]
Yes, some requirements will use it.

**Comment by vincentbernat** [comment]
To implement that, I would need some way to test that. How such an interface appears for Linux? Could you share the output of `ip -d l l dev XXX`?

**Comment by rogerogerogeroger** [comment]
Sorry, there is no actual environment at present, just have this demand. ：）
After that, I will update the information. Thanks！！！

**Comment by rogerogerogeroger** [comment]
Hi, Vincent! I recently docked such a device：[LAN Switching Configuration Guide](http://www.h3c.com.hk/Technical_Support___Documents/Technical_Documents/Routers/H3C_SR6600-X_Series_Routers/Configure/Configuration_Guide/H3C_SR6600_SR6600-X_CG-R7607-6W100/04/201704/990023_1285_0.htm)
It can set the encapsulation format of LLDP frames to SNAP. The result of my capture is as follows：
![2019 09 03_15:20:1567495216_1442x316](https://user-images.githubusercontent.com/24425335/64153894-5cc45b80-ce62-11e9-8a6f-b3b66ab88950.png)
I tried to expand the SNAP function based on your project, can I push the changes up and ask you to review the code for me?
